cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -146,7 +146,7 @@ type Member struct {
 }
 
 // AddMember notifies an existing etcd cluster that a new member is joining
-func (c Client) AddMember(name string, peerAddrs string) ([]Member, error) {
+func (c *Client) AddMember(name string, peerAddrs, clientAddrs string) ([]Member, error) {
```

Comment:
This is for making `AddMember` automatically modify the client endpoints without having to reach anything externally (etcd or the apiserver). However I don't like this because it's not really needed to pass the client address when adding a member. Also, feels hackish to compute this inside the `AddMember` function because the peer address is a complete URI (with scheme and port), and this function could compute it but would have to do tricky operations with it to set the appropriate port and we don't have the init config here to call to `GetClientURL`.

Honestly, if this is too bad maybe we can go back to the alternative option where `AddMember` does not change the client and we manually add the endpoint on `CheckLocalEtcdClusterStatus`.